spec/project/select-modules-in-sources.spec.ts

```diff
@@ -207,6 +254,85 @@ describe('selectModulesInProjectSource', () => {
+                    type IciDocument
+                        @rootEntity #(permissionProfile: "users")
+                        @modules(in: ["module1", "module2"], includeAllFields: true) {
+                        id: ID @key
+
+                        accessGroup: String
+
+                        name: String
+                        mimeType: String
+
+                        "The base64 encoded binary contents"
+                        body: String
+                    }
+
+                    type InternationalCustomsData
+                        @suppress(warnings: UNUSED)
+                        @entityExtension
+                        @modules(in: ["module1", "module2"], includeAllFields: true) {
+                        requestData: InternationalCustomsRequestData
+                        resultData: InternationalCustomsResultData
+                    }
+
+                    type InternationalCustomsRequestData
+                        @entityExtension
+                        @modules(in: ["module1", "module2"], includeAllFields: true) {
+                        isInternationalCustomsRelevant: Boolean
+                            @defaultValue(value: false)
+                            @suppress(infos: NO_TYPE_CHECKS)
+                    }
+
+                    type InternationalCustomsResultData
+                        @entityExtension
+                        @modules(in: ["module1", "module2"], includeAllFields: true) {
+                        internationalCustomsTransmittedTS: DateTime
+                        iciConsignmentId: String
+                        iciDeclarationId: String
+                        customsRegistrationNumber: String
+                        attachments: [IciAttachments]
+                    }
+
+                    type IciAttachments
+                        @valueObject
+                        @modules(in: ["module1", "module2"], includeAllFields: true) {
+                        attachmentId: ID
+                        attachment: IciDocument @reference(keyField: "attachmentId")
+                    }
+
+                    # Used by app ici-result-sync to control sync status with ici.
+                    type SyncId
+                        @rootEntity
+                        @modules(in: ["module1", "module2"], includeAllFields: true) {
+                        systemKey: String @key
+                        syncId: String
+                    }
+                `,
+                [],
+                { removeModuleDeclarations: true },
+            );
+            expect(result).to.equal(`
+                    type OneAndTwo @rootEntity  {
+                        all: String 
+                        two: String 
+                        extra1: String 
+                        
+                    }
+
+                    type Two @rootEntity  {
+                        all: String 
+                        one: String 
+                    }
```

Comment:
Where do those types come from?

Reply:
oops forgot to remove this test case. This is the repro that helped me discover the issue. I replicated the behavior in a simpler case in "removes files that become comment-only" above.

removed it.